storage/innobase/mtr/mtr0mtr.cc

```diff
@@ -983,12 +995,16 @@ static mtr_t::page_flush_ahead log_close(lsn_t lsn) noexcept
       checkpoint_age != lsn)
     log_overwrite_warning(lsn);
   else if (UNIV_LIKELY(checkpoint_age <= log_sys.max_modified_age_async))
-    return mtr_t::PAGE_FLUSH_NO;
-  else if (UNIV_LIKELY(checkpoint_age <= log_sys.max_checkpoint_age))
-    return mtr_t::PAGE_FLUSH_ASYNC;
+    return 0;
+  else
+  {
+    int64_t delta= int64_t(checkpoint_age - log_sys.max_checkpoint_age);
+    if (UNIV_LIKELY(delta <= 0))
+      return lsn - checkpoint_age - delta;
+  }
 
   log_sys.set_check_for_checkpoint();
-  return mtr_t::PAGE_FLUSH_SYNC;
+  return mtr_t::FLUSH_AHEAD_FURIOUS;
```

Comment:
The target LSN in this case should be such that the target age is at most `log_sys.max_modified_age_async`.
Here, the target age is being set as 0 instead (as old code did), and this causes the log to be fully flushed instead.

Reply:
Yes basically that's the idea from https://github.com/MariaDB/server/commit/499e538f317f91dd15cf04b517a40b1e5b9276a8 which made the "furious" case less impactfull. There may be no need to fully flushing the log especially if it's size is not negligible, just flush it to a "safe enough" LSN. But it would also be OK to leave the emergency case as-is and just add a warning log in case the "furious" flushing happens, so that the user can decide if it's necessary to resize the log and avoid same occurrence in the future. Or both.